### PR TITLE
dracut: implement diskless-zram

### PR DESCRIPTION
**Pull request by mischief**
rootfstype=zram will give you ext4 backed by zram on /, with the size set to the size of memory.

### CODE REVIEW

**Comment by mischief** [comment]
cc @marineam @vcaputo

**Comment by crawford** [comment]
What does performance look like between this and in-memory btrfs?

**Comment by vcaputo** [comment]
Looks fine to me, and neat

**Comment by mischief** [comment]
one benefit of this over rootfstype=btrfs w/ lzo is that it will support selinux.

File: dracut/10diskless-generator/diskless-generator [hunk 1]
```diff
@@ -104,6 +104,38 @@ RemainAfterExit=yes
...
 EOF
 
+elif [[ -z "${root}" && "${rootfstype}" == zram ]]; then
+    add_requires sysroot.mount
+    cat >"${UNIT_DIR}/sysroot.mount" <<EOF
+# Automatically generated by diskless-generator
+
+[Unit]
+Requires=diskless-zram.service
+After=diskless-btrfs.service
```
Comment (marineam) [inline dracut/10diskless-generator/diskless-generator]: s/btrfs/zram/

**Comment by marineam** [comment]
The more apt comparison is between this and tmpfs, why is ext4 in ram better than tmpfs?

**Comment by marineam** [comment]
Note: it is possible to use a zram device as swap, also it is possible to do so at any time so that can be done in the normal system. Would doing that allow unused data in tmpfs to be compressed? I don't recall if tmpfs will swap or not since it is backed by the page cache and normally it doesn't make sense to swap the page cache. Memory is fuzzy...

**Pull request closed without merging**